Mathlib/Analysis/Normed/Field/Basic.lean

```diff
@@ -626,7 +626,7 @@ instance MulOpposite.instNormedCommRing : NormedCommRing αᵐᵒᵖ where
   __ := instSeminormedCommRing
 
 /-- The restriction of a power-multiplicative function to a subalgebra is power-multiplicative. -/
-theorem IsPowMul.restriction {R S : Type*} [NormedCommRing R] [CommRing S] [Algebra R S]
+theorem IsPowMul.restriction {R S : Type*} [NormedCommRing R] [Semiring S] [Algebra R S]
```

Comment:
This should be `Ring S`, since the fact `R` is a ring means that `S` can never only be a semiring.

Reply:
Can you explain why `Ring S` is better than `Semiring S`? I agree that in all applications `S` will have to be a ring because of a mathematical argument. What I'm unclear about is why you're saying "make the theorem harder to apply even though we can mathematically prove that ultimately there's no difference". I know we talked about this before on Zulip but since then I've got confused about the logic here. The mathematical argument is problematic for typeclass inference but I'm unclear whether this is relevant. Why is making it a `Ring` making the library better than making it a `Semiring`?